### PR TITLE
Update sidebar v0.2 + nested menu and remove history API

### LINKED ISSUE #24382: Investigate using `<dialog>` `aria-modal` for sidebar 1.0
The current sidebar traps focus (both keyboard and screen reader) when it opens. This is done by adding `aria-hidden` and setting negative tab indicies on things outside of the dialog. Note that it does not use a focus trap that cycles back to the start, so that the correct behavior of focus leaving the browser window when swiping past the start/end is preserved.

The former case can be simplified using `aria-modal` and `role="dialog"` on the sidebar instead of adding `aria-hidden` to other elements. Support for various screen readers here is not exactly clear, but using this approach would simplify the code.

Another approach would be to wrap the contents of the sidebar using a `<dialog>` element. There are a few complications here for styling / animation, due to how the dialog element opens and positions itself, but this handles both screen readers and keyboard focus. This change cannot be done for the existing sidebar due to how things are structured and the way it can be customized, but is potentially something we could do for a new version of sidebar.

We would still need some polyfill behavior, as not all browsers support the dialog element.

/cc @leafsy @cathyxz

### PR DESCRIPTION
**Pull request by leafsy**
I2I: #25049 

resolves #24382, #24668 

This PR makes the following updates to support the new version of `amp-sidebar`:
- Change version from 1.0 to 0.2;
- Change name of nested menu component to `<amp-nested-menu>`;
- Remove history API from nested menu;
- Add unit tests for nested menu;
- Improve a11y for both components by using role="dialog" and shifting focus;

### CODE REVIEW

File: extensions/amp-sidebar/0.2/amp-sidebar.js [hunk 1]
```diff
@@ -222,20 +219,7 @@ export class AmpSidebar extends AMP.BaseElement {
           'A'
         );
         if (target && target.href) {
-          const tgtLoc = Services.urlForDoc(element).parse(target.href);
```
Comment (cathyxz) [inline extensions/amp-sidebar/0.2/amp-sidebar.js]: We should discuss this before removing.

File: extensions/amp-sidebar/0.2/amp-sidebar.js [hunk 2]
```diff
@@ -443,10 +429,6 @@ export class AmpSidebar extends AMP.BaseElement {
       this.element,
       this.getRealChildren()
     );
-    if (this.historyId_ != -1) {
```
Comment (cathyxz) [inline extensions/amp-sidebar/0.2/amp-sidebar.js]: Let's discuss / clarify if we do intend to remove history manipulation from sidebar itself period.

Comment (cathyxz) [reply extensions/amp-sidebar/0.2/amp-sidebar.js]: Per discussion, let's leave it in for now, and you can reference https://github.com/ampproject/amphtml/issues/25080 in a `TODO` to remove or attribute-gate the history manipulation code pending the resolution of that issue.

File: src/sanitation.js [hunk 3]
```diff
@@ -184,10 +184,10 @@ export const WHITELISTED_ATTRS = [
   'subscriptions-display',
   'subscriptions-section',
   'subscriptions-service',
-  // Attributes for amp-drilldown.
-  'amp-drilldown-submenu',
-  'amp-drilldown-submenu-open',
-  'amp-drilldown-submenu-close',
+  // Attributes for amp-nested-menu.
```
Comment (cathyxz) [inline src/sanitation.js]: @choumx for owners bot? 
This is a strict renaming of an unlaunched component.

Comment (choumx) [reply src/sanitation.js]: LGTM

**Pull request merged**